tests/python/test_solana.py

```diff
@@ -1,67 +1,227 @@
 from ragger.backend import RaisePolicy
-from ragger.navigator import NavInsID
+from ragger.navigator import NavInsID, NavIns
 from ragger.utils import RAPDU
 
 from .apps.solana import SolanaClient, ErrorType
-from .apps.solana_cmd_builder import SystemInstructionTransfer, Message, verify_signature
-from .apps.solana_utils import FOREIGN_PUBLIC_KEY, FOREIGN_PUBLIC_KEY_2, AMOUNT, AMOUNT_2, SOL_PACKED_DERIVATION_PATH, SOL_PACKED_DERIVATION_PATH_2
+from .apps.solana_cmd_builder import SystemInstructionTransfer, Message, verify_signature, OffchainMessage
+from .apps.solana_utils import FOREIGN_PUBLIC_KEY, FOREIGN_PUBLIC_KEY_2, AMOUNT, AMOUNT_2, SOL_PACKED_DERIVATION_PATH, SOL_PACKED_DERIVATION_PATH_2, ROOT_SCREENSHOT_PATH
+from .apps.solana_utils import enable_blind_signing, enable_short_public_key, enable_expert_mode, navigation_helper_confirm, navigation_helper_reject
 
-from .utils import ROOT_SCREENSHOT_PATH
 
-def test_solana_simple_transfer_ok_1(backend, navigator, test_name):
-    sol = SolanaClient(backend)
-    from_public_key = sol.get_public_key(SOL_PACKED_DERIVATION_PATH)
+class TestGetPublicKey:
 
-    # Create instruction
-    instruction: SystemInstructionTransfer = SystemInstructionTransfer(from_public_key, FOREIGN_PUBLIC_KEY, AMOUNT)
-    message: bytes = Message([instruction]).serialize()
+    def test_solana_get_public_key_ok(self, backend, navigator, test_name):
+        sol = SolanaClient(backend)
+        from_public_key = sol.get_public_key(SOL_PACKED_DERIVATION_PATH)
 
-    with sol.send_async_sign_message(SOL_PACKED_DERIVATION_PATH, message):
-        navigator.navigate_until_text_and_compare(NavInsID.RIGHT_CLICK,
-                                                  [NavInsID.BOTH_CLICK],
-                                                  "Approve",
-                                                  ROOT_SCREENSHOT_PATH,
-                                                  test_name)
+        with sol.send_public_key_with_confirm(SOL_PACKED_DERIVATION_PATH):
+            if backend.firmware.device.startswith("nano"):
+                navigator.navigate_until_text_and_compare(NavInsID.RIGHT_CLICK,
+                                                          [NavInsID.BOTH_CLICK],
+                                                          "Approve",
+                                                          ROOT_SCREENSHOT_PATH,
+                                                          test_name)
+            else:
+                navigator.navigate_and_compare(ROOT_SCREENSHOT_PATH,
+                                               test_name,
+                                               [NavInsID.USE_CASE_ADDRESS_CONFIRMATION_CONFIRM])
```

Comment:
I thin you should add `NavInsID.USE_CASE_STATUS_DISMISS` at the end of all stax flows so thatt they are correctly ended.